### DISCUSSION TITLE
Write to USB.

### DISCUSSION THREAD

**Discussion by ikeysolomon** [aaronwmorris/indi-allsky#1274] [General]
Hi everyone, 
Just wondering about how you would have it write the images to a usb?
Thanks 
Sam

**Comment by aaronwmorris** [comment]
There is a script in the misc folder to setup USB automounting...

```
./misc/setup_usb_automount.sh
```

When you run the script, it will prompt you to insert the USB stick.  The mounted device should be somewhere under `/media/username/`  You can then go into the web interface and change the image folder on the `Image` tab about half-way down.  Then restart indi-allsky.

Just a warning, at this point, the USB drive needs to be mounted for indi-allsky to operate properly.

**Reply by ikeysolomon** [reply]
Thank you. Worked a treat. However any scope to incorporate the last view when the images are written to usb? It worked OK but it lost that function.

**Reply by aaronwmorris** [reply]
I forgot a step, you just need to re-run setup.sh to fix the flask config.  It just needs the image folder updated which will be automaically updated with setup.sh

**Comment by ikeysolomon** [comment]
Reference using writing to USB. Is there a means to change the "batch" size, I.e. ive noticed that the software wants to place X amount of images in a folder. Can I configure it such that it's all just in the one folder?

**Reply by aaronwmorris** [reply]
Unfortunately, no.  The files are split into so many folders for performance reasons.  Once you get a few thousand files in a single folder, accessing files and folder can start to degrade significantly.  If you take images ever 15 seconds like I do, you end up with over 5,000 images per day.  Double that if you have panoramas.  Thumbnails add even more.

Is there some way you want to access files?  Due to all of the files being referenced in the database, it makes it very easy to generate lists of files.  There are ways to build lists on the command line, as well.  Maybe I can come up with something for you.

If you want to manually hack this, the function is `_getImageFolder()` here:  https://github.com/aaronwmorris/indi-allsky/blob/main/indi_allsky/image.py#L1398

**Reply by ikeysolomon** [reply]
Thanks for getting back to me. I'll provide some context. I've been writing to a usb. And then transferring them to my computer and stitching them together using pip. So many opportunities for improvement here I know but my learnings only go so far at this point. With time I will read more on network options for transfer. So for me going in and out of each folder is a bit untidy to copy. As I said I intend to read up and learn more but this what I know and what works at this point.
